shared/utils/typedefs-string.hpp

```diff
@@ -83,13 +86,25 @@ struct ConstString {
         return {chars, static_cast<std::size_t>(sz)};
     }
 
+    friend StringW;
     private:
     void* klass = nullptr;
     void* monitor;
     int length;
     char16_t chars[sz + 1];
 };
 
+bool streq(Il2CppString const* lhs, const std::string_view rhs) noexcept;
```

Comment:
Can you move this (and all the other function declarations) to the `il2cpp_utils::detail` namespace, see elsewhere in this file.

Reply:
Also east const on pointers, yay! west const on other types, awww :(

Reply:
Sorry what do you mean with West const on other types?